src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -268,6 +268,11 @@ public abstract class $Type$Buffer
     extends Buffer
     implements Comparable<$Type$Buffer>{#if[char]?, Appendable, CharSequence, Readable}
 {
+    // Cached array base offset
+    private static final long ARRAY_BASE_OFFSET = UNSAFE.arrayBaseOffset($type$[].class);
+
+    // Number of bytes per $type$
+    private static final long ELEMENT_SIZE = 1L << $LG_BYTES_PER_VALUE$;
```

Comment:
IIRC you could use the constant `$Type$.BYTES` rather than declare a new static field.

Reply:
Yes, you are correct; will update.